src/components/LabelWarning.vue

```diff
@@ -0,0 +1,163 @@
+<script>
+import { mapGetters } from 'vuex'
+export default {
+  props: {
+    flow: {
+      type: Object,
+      default: () => {}
+    },
+    flowGroup: {
+      type: Object,
+      default: () => {}
+    },
+    iconSize: {
+      type: String,
+      required: false,
+      default: null
+    },
+    location: {
+      type: String,
+      default: ''
+    }
+  },
+  data() {
+    return {
+      infoMessage: ''
+    }
+  },
+  computed: {
+    ...mapGetters('tenant', ['tenant']),
+    ...mapGetters('agent', ['agents']),
+    agentOrLabel() {
+      if (!this.agents || !this.agents.length) return 'Agent Problem'
+      if (!this.labelsAlign) return 'Label Problem'
+      return 'Flow and Agent Labels'
+    },
+    agentLabels() {
+      return this.agents?.reduce((accumulator, agent) => {
+        accumulator.push(agent.labels)
+        return accumulator
+      }, [])
+    },
+    flowLabels() {
+      return this.flowGroup?.labels || this.flow?.environment?.labels
+    },
+    docsName() {
+      if (!this.agents?.length) return 'agents'
+      return 'Labels and Flow Affinity'
+    },
+    docsLink() {
+      if (!this.agents?.length)
+        return 'https://docs.prefect.io/orchestration/agents/overview.html'
+      return 'https://docs.prefect.io/orchestration/agents/overview.html#flow-affinity-labels'
+    },
+    labelsAlign() {
+      if (!this.agents?.length) {
+        this.labelMessage(
+          'You have no live Agents - scheduled flow runs will not be submitted for execution and will display as "Late".'
+        )
+        return false
+      }
+      if (
+        !this.flowLabels?.length &&
+        this.agentLabels.every(arrayOfLabels => arrayOfLabels.length > 0)
+      ) {
+        this.labelMessage(
+          'You have no Agents configured to pick up flows without labels; you may need to add labels to your flow.'
+        )
+        return false
+      } else {
+        let matchingLabels = 0
+        if (this.agentLabels) {
+          this.agentLabels.forEach(array => {
+            if (this.flowLabels.every(label => array.includes(label)))
+              matchingLabels++
+          })
+        }
+        if (matchingLabels > 0) {
+          this.labelMessage('Your flow and agent labels look good.')
+          return true
+        } else {
+          this.labelMessage(
+            'It looks like you have a mismatch of labels between your flow and running Agents. To allow an Agent to pick up this flow run, you need to have at least one Agent whose labels include those on the flow.'
```

Comment:
```suggestion
            'It looks like no currently running Agent has this flow's full set of labels. To allow an Agent to run this flow, you need to have at least one Agent whose labels include all of those on the flow.'
```

Reply:
The wording proposal is hopefully more agnostic to whether the warning is displayed on a flow run or flow tile, and is slightly shorter